src/sqlancer/yugabyte/ysql/YSQLProvider.java

```diff
@@ -0,0 +1,366 @@
+        }
+
+        return con;
+    }
+
+    protected void readFunctions(YSQLGlobalState globalState) throws SQLException {
+        SQLQueryAdapter query = new SQLQueryAdapter("SELECT proname, provolatile FROM pg_proc;");
+        SQLancerResultSet rs = query.executeAndGet(globalState);
+        while (rs.next()) {
+            String functionName = rs.getString(1);
+            Character functionType = rs.getString(2).charAt(0);
+            globalState.addFunctionAndType(functionName, functionType);
+        }
+    }
+
+    protected void createTables(YSQLGlobalState globalState, int numTables) throws Exception {
+        synchronized (CREATION_LOCK) {
+            boolean prevCreationFailed = false; // small optimization - wait only after failed requests
+            while (globalState.getSchema().getDatabaseTables().size() < numTables) {
+                if (!prevCreationFailed) {
+                    exceptionLessSleep(5000);
+                }
+
+                try {
+                    String tableName = DBMSCommon.createTableName(globalState.getSchema().getDatabaseTables().size());
+                    SQLQueryAdapter createTable = YSQLTableGenerator.generate(tableName, generateOnlyKnown,
+                            globalState);
+                    globalState.executeStatement(createTable);
+                    prevCreationFailed = false;
+                } catch (IgnoreMeException e) {
+                    prevCreationFailed = true;
+                }
+            }
+        }
+    }
+
+    private void exceptionLessSleep(long timeout) {
+        try {
+            Thread.sleep(timeout);
+        } catch (InterruptedException e) {
+            e.printStackTrace();
+        }
+    }
+
+    protected void prepareTables(YSQLGlobalState globalState) throws Exception {
+        StatementExecutor<YSQLGlobalState, Action> se = new StatementExecutor<>(globalState, Action.values(),
+                YSQLProvider::mapActions, (q) -> {
+                    if (globalState.getSchema().getDatabaseTables().isEmpty()) {
+                        throw new IgnoreMeException();
+                    }
+                });
+        se.executeStatements();
+        globalState.executeStatement(new SQLQueryAdapter("COMMIT", true));
+        globalState.executeStatement(new SQLQueryAdapter("SET SESSION statement_timeout = 15000;\n"));
+    }
+
+    private String getCreateDatabaseCommand(YSQLGlobalState state) {
+        StringBuilder sb = new StringBuilder();
+        sb.append("CREATE DATABASE ").append(databaseName).append(" ");
+        if (Randomly.getBoolean() && state.getDbmsSpecificOptions().testCollations) {
+            sb.append("WITH ");
+            if (Randomly.getBoolean()) {
+                sb.append("ENCODING '");
+                sb.append(Randomly.fromOptions("utf8"));
+                sb.append("' ");
+            }
+            // disabled due to https://github.com/yugabyte/yugabyte-db/issues/11357
```

Comment:
Here, we could also apply the `TiDBBugs` pattern.

Reply:
Added YugabyteBugs, currently with one one issue, will extend it bit later